scripts/add_directory.js

```diff
@@ -0,0 +1,289 @@
+// set that contains all of the folders currently added as batches
+let batchDirs = new Set();
+// reference to the autocomplete object
+let autocomplete;
+// iterator used to give all batch UI's distinct ID's
+let i = 0;
+// reference to fadeOut timeout for the resultText, used for cancelling the timeout if another result text needs to
+// be shown before the last result text has faded
+let fadeTimeout;
+
+$(function() {    
+    // matches strings for top level directories
+    let parentDirReg = /^\w+\/$/;
+    $.getJSON("scripts/folders.json").done(function(dirObject) {
+        let input = document.getElementById('folder-input');
+        autocomplete = new Awesomplete(input, {
+            list: dirObject.folders,
+            autoFirst: true,
+            filter: function(text, input) {
+                if (batchDirs.has(text.value)) {
+                    return false;
+                }
+                // if input is empty, list top level dirs
+                if (input.length == 0) {
+                    return parentDirReg.test(text.label);
+                }
+                else {
+                    // default filter function
+                    return Awesomplete.FILTER_CONTAINS(text, input);
+                }
+            },
+            minChars: 0
+        });
+        $(input).focus(function(ev) {
+            autocomplete.evaluate();
+        });
+    });
+
+    $("#add-directory-form").submit(function(ev) {
+        ev.preventDefault();
+        let dir = $("#folder-input").val()
+        resultText(dir, addBatch(dir));
+    })
+});
+
+function addBatch(dir) {
+    if (dir.length > 0 && !batchDirs.has(dir)) {
+        batchDirs.add(dir);
+        let newBatch = $(createBatchUi(dir, i++));
+        $("#add-directory-wrapper").append(newBatch);
+        newBatch.fadeIn();
+        return true;
+    }
+    return false;
+}
+
+function resultText(dir, suc) {
+    // displays text when one batch is added
+    let resText = $("#batch-response");
+    let textClass, text;
+    if (suc) {
+        text = dir + " added as batch";
+        textClass = 'text-success';
+    }
+    // dont print out empty dir names
+    else if (dir.length > 0) {
+        text = dir + " already exists";
+        textClass = "text-warning";
+    }
+    else {
+        return;

```

Comment:
Wait... does this just return nothing if the addbatch fails? I might be reading the usage of this incorrectly. It might be helpful to explain what failed.